creator-node/src/routes/tracks.js

```diff
@@ -37,12 +40,90 @@ const RehydrateIpfsQueue = require('../RehydrateIpfsQueue')
 const { FileProcessingQueue } = require('../FileProcessingQueue')
 const DBManager = require('../dbManager')
 const { generateListenTimestampAndSignature } = require('../apiSigning.js')
+const DiskManager = require('../diskManager')
 
+const server = new tus.Server()
 const readFile = promisify(fs.readFile)
+const tmpTrackArtifactsPath = DiskManager.getTmpTrackUploadArtifactsPath()
+const resumableUploadRoute = path.join(tmpTrackArtifactsPath, '*', '*')
+
+// Structure: <uuid>.<file_extension>. e.g. 1234-1234-1234-1234.mp3
+const getFileName = (req) => {
+  return req.headers.randomfilename + getFileExtension(req.headers.filename)
+}
 
 const SaveFileToIPFSConcurrencyLimit = 10
 
 module.exports = function (app) {
+  app.post('/track_content_upload', authMiddleware, ensurePrimaryMiddleware, ensureStorageMiddleware, syncLockMiddleware, checkFileMiddleware, async function (req, res, next) {
+    // Use the tus-node-server package to handle track upload
+
+    // Save file under randomly named folders to avoid collisions
+    const fileDir = path.join(tmpTrackArtifactsPath, req.headers.randomfilename)
+
+    try {
+    // Create directories for original file and segments
+      fs.mkdirSync(fileDir, { recursive: true })
```

Comment:
you should be able to call DiskManager.ensureDirPathExists(fileDir) for this. also we shouldn't be using the "sync" version of these functions. if we use the fs-extra module we get async/await and it doesn't block the event loop